src/segment-loader.js

```diff
@@ -162,14 +162,10 @@ const segmentInfoString = (segmentInfo) => {
 
   const hasPartIndex = typeof partIndex === 'number';
   const name = segmentInfo.segment.uri ? 'segment' : 'pre-segment';
-  const partCount = segment.parts ? segment.parts.length : 0;
-  const preloadPartCount = segment.preloadHints ?
-    segment.preloadHints.filter((h) => h.type === 'PART').length :
-    0;
-  const zeroBasedPartCount = partCount + preloadPartCount - 1 - (preloadPartCount > 0 ? 1 : 0);
+  const totalParts = hasPartIndex ? getKnownPartCount({preloadSegment: segment}) - 1 : 0;
 
   return `${name} [${seq + index}/${seq + segmentLen}]` +
-    (hasPartIndex ? ` part [${partIndex}/${zeroBasedPartCount}]` : '') +
+    (hasPartIndex ? ` part [${partIndex}/${totalParts}]` : '') +
```

Comment:
Should this take into account the 0 based indexing vs total. Maybe we can just say `[index 0, 3 total]

Reply:
I think we should keep the name `zeroBasedPartCount` here I guess. I do subtract 1 to make it zero based.